dist/svg.js

```diff
@@ -553,16 +553,28 @@ SVG.extend(SVG.PointArray, {
     // if already is an array, no need to parse it
     if (Array.isArray(array)) return array
 
-    // split points
-    array = this.split(array)
 
     // parse points
-    for (var i = 0, il = array.length, p, points = []; i < il; i++) {
-      p = array[i].split(',')
-      points.push([parseFloat(p[0]), parseFloat(p[1])])
-    }
+    var points = array.split(/\s|,/)
+
+    // validate points - https://svgwg.org/svg2-draft/shapes.html#DataTypePoints
+    // Odd number of coordinates is an error. In such cases, drop the last odd coordinate.
+    if (points.length % 2 !== 0) points.pop()
+
+    // parse points as floats
+    points = points.map(function(x) { return parseFloat(x) })
+
+    // wrap points in two-tuples
+    points = points.map(wrapPoints()).filter(function(x) { return x })
```

Comment:
As long as we don't have more higher order FP functions we can use, I agree that list manipulations quickly grow a bit hairy. I'll change it to a `for loop`.